simple_mediation.R

```
@@ -1,86 +1,58 @@
-df_males_na_removed$mother_race <- as.factor(df_males_na_removed$mother_race)
-df_males_na_removed$father_race <- as.factor(df_males_na_removed$father_race)
-df_males_na_removed$mental_health <- as.factor(df_males_na_removed$mental_health)
-
-df_males_na_removed$model3_trajectory_assignments <- as.factor(df_males_na_removed$model3_trajectory_assignments) 
-#Run analysis 
-df_males_na_removed[,"prosocial"] %>% 
-  mutate_at(("prosocial"), ~(scale(.) %>% as.vector)) -> df_males_scaled
-
-#expFit <- glm(model3_trajectory_assignments*SEP ~ Remoteness + Indigenous + adv + dis + resources + education_occupation + mother_home + father_home + mental_health + mother_race + father_race, data = df_males_na_removed)
-expData <- neImpute(prosocial ~ SEP + model3_trajectory_assignments + Remoteness + Indigenous + adv + dis + resources + education_occupation + mother_home + father_home + mental_health + mother_race + father_race, data = df_males_na_removed)
-neProsocial <- neModel(prosocial ~ SEP0 + SEP1 + Remoteness + Indigenous + adv + dis + resources + education_occupation + mother_home + father_home + mental_health + mother_race + father_race, expData = expData, se = "robust")
-summary(neProsocial)
-str(neProsocial)
-
-#I trired to "unlist" the results but this did not solve the error
-#unlist(neProsocial) -> neProsocial2
-head(neProsocial)
-
-effdecomp <- neEffdecomp(neProsocial)
-summary(effdecomp)
-
-summary(expData)
-head(expData)
-expData2 <- neImpute(prosocial ~ SEP + model3_trajectory_assignments + Remoteness + Indigenous + adv + dis + resources + education_occupation + mother_home + father_home + mental_health + mother_ethnicity + father_ethnicity, family = poisson(link = "log"), data = df_males_na_removed)
\ No newline at end of file
+library(rmarkdown)
+
+source("./load_data_by_sex.R")
+source("./munging.R")
+male_data()
+munge("male")
+female_data()
+munge("female")
+
+#Run loop for all socio-emotional variables
+for (x in 16:ncol(df_male)) {
```

Be a bit careful with `16:ncol()`, because if you change something upstream which affects the number of columns this will fail silently, and is pretty hard to debug.

A list of column names can be safer.